libnetdata/url/url.c

```diff
@@ -52,15 +52,78 @@ char *url_decode(char *str) {
     return url_decode_r(buf, str, size);
 }
 
+//decode %XX character or return 0 if cannot
+char url_percent_escape_decode(char *s) {
+    if(likely(s[1] && s[2]))
+        return from_hex(s[1]) << 4 | from_hex(s[2]);
+    return 0;
+}
+
+//this (utf8 string related) should be moved in separate file in future
+char url_utf8_get_byte_length(char c) {
+    if(!IS_UTF8_BYTE(c))
+        return 1;
+
+    char length = 0;
+    while(likely(c & 0x80)) {
```

Comment:
i did ctrl-f `& 0x80` and found several places where you can use `IS_UTF8_BYTE`

Reply:
Well for example in fucntion url_utf8_get_byte_length it is on purpose
because there i do not check is it UTF8 char but in cycle i take MSB bit out and count it.... so the condition is also c & 0x80 but meaning why is different

Reply:
Nice, thanks for the explanation!